### DISCUSSION TITLE
sweep_along_path （）Generates a model name error

### DISCUSSION THREAD

**Discussion by lixiangru123** [ansys/pyaedt#4938] [Q&A] [answered]
spiral = Spiral_equalizer(peq_cent_x, peq_cent_y,   s_spiral,  n_spiral, rin_spiral,h_spiral_line, d_clearance,h_sub) 
        spiral_line_point = spiral.generate_line()

        Spiral_line = hfss.modeler.create_polyline(position_list=spiral_line_point,name='spiral_line')
        Spiral_rectangle = hfss.modeler.create_rectangle("XY", [peq_cent_x-0.5*h_gnd,peq_cent_y-rin_spiral-0.5*w_spiral,h_sub+h_gnd],
                                                         [h_gnd,w_spiral], name='Spiral_EQ', matname='copper')
          hfss.modeler.sweep_along_path(objid=Spiral_line,sweep_object=Spiral_rectangle) #? BUG  Run here and the model pops up an error alert，But it doesn't affect modeling

![64835eeaee04dab5b6fdd5e12ea3d12e](https://github.com/user-attachments/assets/4e6325cf-7ffb-4d1c-ade9-d3cece8b1b36)
![be3f0ff0-dbfd-4e4e-91b1-674b163f4844](https://github.com/user-attachments/assets/5840eb55-e9b8-44b9-868b-a84010bb5070)

**Comment by lixiangru123** [comment]
This issue does not affect emulation in the software, but causes the following error when I use hfss.export_touchstone(file_name=r'd :\snp\spiral.s6p',sweep='sweep1',setup='Spiral_setup')
![e6deed18-7111-45d7-960f-32931069470e](https://github.com/user-attachments/assets/80652b7f-4164-4327-b35b-1d6ca896e71e)
Because my final task is to export the snp file

**Comment by lixiangru123** [comment]
I can now export the snp file normally, but the error alert of the model is still not resolved, even though the entire error does not affect use.

**Comment by gmalinve** [comment]
Hi @lixiangru123,

Thanks for opening this discussion. Could you please provide the whole code snippet where you create the polyline, the rectangle and then you try to sweep please along path?
I'd be interested to see the parametrized list of points _spiral_line_point_ and the arguments you are passing to the _create_rectangle_ (specifically these: _[peq_cent_x-0.5*h_gnd,peq_cent_y-rin_spiral-0.5*w_spiral,h_sub+h_gnd], [h_gnd,w_spiral]_).

It seems that the polyline you are trying to create is not correctly created.

Thanks!
Kind regards,

Giulia

**Reply by lixiangru123** [reply]
If you have time, I uploaded the complete code of the model on my home page, and invite you to join, cancel the last comment to run, PyAEDT version ==0.6.71 version.

**Comment by gmalinve** [comment]
I also forgot to mention that you are still using old arguments in the methods.
For now is not a problem, these only issue warnings but with PyAEDT version 1.0 these will become breaking changes.
Please look here: https://aedt.docs.pyansys.com/version/stable/release_1_0.html#release-1-0.
@lixiangru123

**Comment by lixiangru123** [comment]
```      # line1
        line1_point = [(-w1, 0, 2 * h + 0.5 * h),
                       (-w1, 0.5 * a, 2 * h + 0.5 * h),
                       (-(0.5 * s1 + 0.5 * w1), 0.5 * a, 2 * h + 0.5 * h),
                       (-(0.5 * s1 + 0.5 * w1), 0.5 * a + 2 * a + d1, 2 * h + 0.5 * h),
                       (-w1, 0.5 * a + 2 * a + d1, 2 * h + 0.5 * h),
                       (-w1, 0.5 * a + 2 * a + d1 + 0.5 * a, 2 * h + 0.5 * h)]
        line1_line = hfss.modeler.create_polyline(position_list=line1_point, name='line1_line')  
        line1_rectangle = hfss.modeler.create_rectangle("XZ", [-0.5 * w1, 0, h + h_cond], [h_cond,-w1 ], name='line1',matname='copper')
        hfss.modeler.sweep_along_path(objid=line1_line, sweep_object=line1_rectangle)  # ? BUG

          # line2
        line2_point = [(w1, 0, 2 * h + 0.5 * h),
                       (w1, 0.5 * a, 2 * h + 0.5 * h),
                       ((0.5 * s1 + 0.5 * w1), 0.5 * a, 2 * h + 0.5 * h),
                       ((0.5 * s1 + 0.5 * w1), 0.5 * a + 2 * a + d1, 2 * h + 0.5 * h),
                       (w1, 0.5 * a + 2 * a + d1, 2 * h + 0.5 * h),
                       (w1, 0.5 * a + 2 * a + d1 + 0.5 * a, 2 * h + 0.5 * h)]
        line2_line = hfss.modeler.create_polyline(position_list=line2_point, name='line2_line')  
        line2_rectangle = hfss.modeler.create_rectangle("XZ", [0.5 * w1, 0, h + h_cond], [h_cond,w1 ], name='line2',matname='copper')
        hfss.modeler.sweep_along_path(objid=line2_line, sweep_object=line2_rectangle)  # ? BUG
```
![1727348519499](https://github.com/user-attachments/assets/fe6bae8e-1db5-45e0-8bac-a80b817f76fd)


   Hello, thank  @gmalinve  for your kind reply! Indeed, I used a relatively old version of PyAEDT,0.6.71. The previous code was that I drew a quadrilateral spiral line, and then scanned the fold line with a rectangle into a spiral circle. But recently I tried the new model, and the new model is just a line, as the code above shows, and the same warning still comes up. I've also attached a model and a warning.This warning does not affect the actual modeling

![1727348663593](https://github.com/user-attachments/assets/290b1836-16b5-4fed-a6f5-9ef80a7650eb)

![image](https://github.com/user-attachments/assets/2825439f-2443-414c-9021-190712ec98b0)

**Comment by gmalinve** [comment, marked as answer]
Hi @lixiangru123,

Could you please update to latest version?
0.6 is very old, we are now at 0.10.2.
Please update and re-try your workflow.

Thanks,
Regards,

Giulia
